components/server/src/workspace/workspace-starter.ts

```diff
@@ -345,7 +346,7 @@ export class WorkspaceStarter {
                 instance,
                 workspace,
                 user,
-                lastValidWorkspaceInstanceId,
+                lastValidWorkspaceInstance?.id ?? "",
```

Comment:
In TypeScript, passing around "undefined" is the way to denote "not present". So `lastValidWorkspaceInstance?.id` (which is `string | undefined`) is way more in line with expectations than `string` (empty or not).

Reply:
Ah, did not check that. And we even do untyped equality checks against it... :see_no_evil: 
Ok, fixing this is out of scope here. :cry: